src/coordination/coordinator_state_machine.cpp

```diff
@@ -181,47 +181,60 @@ auto CoordinatorStateMachine::SerializeUpdateUUIDForInstance(InstanceUUIDUpdate
   return CreateLog({{"action", RaftLogAction::UPDATE_UUID_FOR_INSTANCE}, {"info", instance_uuid_change}});
 }
 
-auto CoordinatorStateMachine::DecodeLog(buffer &data) -> std::pair<TRaftLog, RaftLogAction> {
+auto CoordinatorStateMachine::DecodeLog(buffer &data) -> std::optional<std::pair<TRaftLog, RaftLogAction>> {
   buffer_serializer bs(data);
-  auto const json = nlohmann::json::parse(bs.get_str());
+  std::string const log_str = bs.get_str();
+  if (log_str.empty()) {
+    return std::nullopt;  // this is returned only when the log is configuration log
+  }
+  auto const json = nlohmann::json::parse(log_str);
```

Comment:
Does this mean we cannot create log with empty string or? Do we always have some string

Reply:
Actually I don't need this but it was used as a safety check if data is empty we have nothing to decode. But that case is handled in commit function in this file so I will remove this

Reply:
I changed this part as I figured out we have commit config function